url933dd5727b_AppleLiang_Testing_Project_tgz-pJ8-org_joda_time_chrono_gj_MainTestJ8/src/org/joda/time/TestDateMidnight_Constructors.java

```diff
@@ -0,0 +1,621 @@
+            DateMidnight test = new DateMidnight(new Integer(0));
+            assertEquals(ISOChronology.getInstance(), test.getChronology());
+            assertEquals(0L - DateTimeConstants.MILLIS_PER_HOUR, test.getMillis());
+        } finally {
+            ConverterManager.getInstance().removeInstantConverter(MockZeroNullIntegerConverter.INSTANCE);
+        }
+    }
+
+    /**
+     * Test constructor (Object, DateTimeZone)
+     */
+    public void testConstructor_Object_DateTimeZone() throws Throwable {
+        Date date = new Date(TEST_TIME1_UTC);
+        DateMidnight test = new DateMidnight(date, PARIS);
+        assertEquals(ISOChronology.getInstance(PARIS), test.getChronology());
+        assertEquals(TEST_TIME1_PARIS, test.getMillis());
+    }
+
+    /**
+     * Test constructor (Object, DateTimeZone)
+     */
+    public void testConstructor_invalidObject_DateTimeZone() throws Throwable {
+        try {
+            new DateMidnight(new Object(), PARIS);
+            fail();
+        } catch (IllegalArgumentException ex) {}
+    }
+
+    /**
+     * Test constructor (Object=null, DateTimeZone)
+     */
+    public void testConstructor_nullObject_DateTimeZone() throws Throwable {
+        DateMidnight test = new DateMidnight((Object) null, PARIS);
+        assertEquals(ISOChronology.getInstance(PARIS), test.getChronology());
+        assertEquals(TEST_TIME_NOW_PARIS, test.getMillis());
+    }
+
+    /**
+     * Test constructor (Object, DateTimeZone=null)
+     */
+    public void testConstructor_Object_nullDateTimeZone() throws Throwable {
+        Date date = new Date(TEST_TIME1_UTC);
+        DateMidnight test = new DateMidnight(date, (DateTimeZone) null);
+        assertEquals(ISOChronology.getInstance(), test.getChronology());
+        assertEquals(TEST_TIME1_LONDON, test.getMillis());
+    }
+
+    /**
+     * Test constructor (Object=null, DateTimeZone=null)
+     */
+    public void testConstructor_nullObject_nullDateTimeZone() throws Throwable {
+        DateMidnight test = new DateMidnight((Object) null, (DateTimeZone) null);
+        assertEquals(ISOChronology.getInstance(), test.getChronology());
+        assertEquals(TEST_TIME_NOW_LONDON, test.getMillis());
+    }
+
+    /**
+     * Test constructor (Object, DateTimeZone)
+     */
+    public void testConstructor_badconverterObject_DateTimeZone() throws Throwable {
+        try {
+            ConverterManager.getInstance().addInstantConverter(MockZeroNullIntegerConverter.INSTANCE);
+            DateMidnight test = new DateMidnight(new Integer(0), GregorianChronology.getInstance());
+            assertEquals(ISOChronology.getInstance(), test.getChronology());
+            assertEquals(0L - DateTimeConstants.MILLIS_PER_HOUR, test.getMillis());
+        } finally {
+            ConverterManager.getInstance().removeInstantConverter(MockZeroNullIntegerConverter.INSTANCE);
+        }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 400:407, 462:469.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.